### ISSUE TITLE
Replay the current media

### ISSUE THREAD

**Issue by caprica** [caprica/vlcj#330]
It used to be the case with LibVLC that if the current media reached a stopped or ended state, that if you invoked play the media would play again.

This is no longer the case.

Invoking play on a stopped or ended media will do nothing, and requires the preparation of a new MRL before playing.

In vlcj, this means that `playMedia(mrl)` must be invoked again, not just `play()`.

Any solution must restore not just the MRL, but any `mediaOptions` that were specified too.

A vlcj application can do that itself anyway, but it might be convenient if vlcj provided a way to automatically do it.

The `start()` method may also need to change to work the same way.

Basically, `prepareMedia()` must be invoked again if the player state is stopped or finished (or error I suppose), or is it just "not paused" ?

**Comment by caprica** [comment]
The problem is unless you rely on events (which the application can do), then there is a small chance of a race.

**Comment by caprica** [comment]
Probably the best thing to do is for the media player implementation to listen for the stopped and finished events, and reset the media.

**Comment by caprica** [comment]
This actually only applies to the finished event, not the stopped event.

**Comment by caprica** [comment]
Completed at 1dfe1adede8b38a6ba1b20f6ecbe1e54c3791e9b.
